### ISSUE TITLE
Support for `UNION`, `INTERSECT` & `EXCEPT` in `SQLContext`

### ISSUE THREAD

**Issue by csubhodeep** [pola-rs/polars#8917]
### Problem description

I wish Polars SQL could support combination of results from multiple queries using set operations. 

Relates to #8743.

References:
1. [Combining Queries (UNION, INTERSECT, EXCEPT)](https://www.postgresql.org/docs/current/queries-union.html)

**Comment by alexander-beedie** [comment]
`UNION` is now done: https://github.com/pola-rs/polars/pull/8936

**Comment by xdev-x** [comment]
Is it possible for UNION to concatenate with the `diagonal` strategy ?

**Comment by cmdlineluser** [comment]
@xdev-x I thought `UNION BY NAME` was supposed to do that.

```python
df1 = pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
df2 = pl.DataFrame({"c": [4, 5, 6, 7]})

pl.SQLContext(frames={"df1": df1, "df2": df2}).execute("""
select * from df1 union by name select * from df2
""").collect()

# ShapeError: unable to append to a dataframe of width 2 with a dataframe of width 1
```

**Comment by alexander-beedie** [comment]
> Is it possible for UNION to concatenate with the `diagonal` strategy ?

It will be in the upcoming `0.19.8`; we have committed support for `UNION BY NAME` and `UNION ALL BY NAME` ✌️

**Comment by alexander-beedie** [comment]
FYI: this was closed by https://github.com/pola-rs/polars/pull/16960 👌 
All three of these set ops will be available in the 1.0 release.
